src/libs/Middleware/HandleUnusedOptimisticID.ts

```diff
@@ -0,0 +1,39 @@
+import _ from 'lodash';
+import ONYXKEYS from '../../ONYXKEYS';
+import Report from '../../types/onyx/Report';
+import {Middleware} from '../Request';
+import * as PersistedRequests from '../actions/PersistedRequests';
+import deepReplaceKeysAndValues from '../deepReplaceKeysAndValues';
+
+const handleUnusedOptimisticID: Middleware = (requestResponse, request, isFromSequentialQueue) =>
+    requestResponse.then((response) => {
+        const responseOnyxData = response?.onyxData ?? [];
+        responseOnyxData.forEach((onyxData) => {
+            const key = onyxData.key;
+            if (!key.startsWith(ONYXKEYS.COLLECTION.REPORT)) {
```

Comment:
`key` could be `undefined` which caused this issue - https://github.com/Expensify/App/issues/28748#issuecomment-1746091428